### DISCUSSION TITLE
Support for client-signature-algorithms

### DISCUSSION THREAD

**Discussion by mohitvaid** [eclipse-paho/paho.mqtt.c#1226] [Ideas]
@icraggs  We are using syslog-ng for MQTT communication. Which internally uses paho.mqtt.c

As part of our NIAP support we are supposed to restrict certain TLS specific parameters, likewise:

- ciphers,
- client signature algorithms/ signature algorithms,
- ecdh curves

Based on https://github.com/syslog-ng/syslog-ng/pull/4000, syslog-ng team has made changes for their general TLS support but for MQTT flow they are unable to as paho library lacks support for the signature algorithms.

Can this change be accommodated?

**Comment by mohitvaid** [comment]
@icraggs  can you please take a look at this discussion?

**Comment by icraggs** [comment]
Sorry I hadn't really been aware of this discussion facility. It seems the best way to support this would be to expose the OpenSSL context for configuration by the application as suggested in the linked discussion. The request actually seems to be to _restrict_ some algorithms.
